src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
As the Spanish always say (probably), "muy sexy." 

I'm sold, seems like a super nice approach. Imma re-re-review the rest of the PR now that you've convinced me on this part.